polar-core/src/kb.rs

```diff
@@ -162,6 +162,37 @@ impl KnowledgeBase {
                 }
             }
         }
+
+        // Enforce rule type validation for "required" rule types derived from
+        // resource block definitions.
+        for rule_type in self.rule_types.required_rule_types() {
+            if let Some(GenericRule { rules, .. }) = self.rules.get(&rule_type.name) {
+                let found_match = rules.values().any(|rule| {
+                    self.rule_params_match(rule.as_ref(), rule_type)
+                        .map(|r| match r {
+                            RuleParamMatch::True => true,
+                            RuleParamMatch::False(_) => false,
+                        })
+                        .unwrap()
```

Comment:
Is this `.unwrap()` cool?

Reply:
I think this should be safe, the reason being
1. the only `Err` case this function can encounter relates checking subtypes for specializers
2. the rule types we're checking here were created with specializer symbols lifted directly from the policy which we require to be registered.

Reply:
we talked about this; gonna try using `?` instead